src/java/org/apache/cassandra/cql3/constraints/ColumnConstraints.java

```diff
@@ -0,0 +1,179 @@
+                                                                                                CompositeType.class,
+                                                                                                DynamicCompositeType.class);
+
+
+
+    private final List<ColumnConstraint<?>> constraints;
+
+    public ColumnConstraints(List<ColumnConstraint<?>> constraints)
+    {
+        this.constraints = new ArrayList<>(constraints);
+    }
+
+    @Override
+    public IVersionedSerializer<ColumnConstraints> serializer()
+    {
+        return serializer;
+    }
+
+    @Override
+    public void appendCqlTo(CqlBuilder builder)
+    {
+        for (ColumnConstraint<?> constraint : constraints)
+            constraint.appendCqlTo(builder);
+    }
+
+    @Override
+    public void evaluate(Class<? extends AbstractType> valueType, Object columnValue) throws ConstraintViolationException
+    {
+        for (ColumnConstraint<?> constraint : constraints)
+            constraint.evaluate(valueType, columnValue);
+    }
+
+    public List<ColumnConstraint<?>> getConstraints()
+    {
+        return constraints;
+    }
+
+    public boolean isEmpty()
+    {
+        return constraints.isEmpty();
+    }
+
+    @Override
+    public void validate(ColumnMetadata columnMetadata) throws InvalidConstraintDefinitionException
+    {
+        if (UNSUPPORTED_TYPES.contains(columnMetadata.type.getClass()))
+            throw new InvalidConstraintDefinitionException("Constraint cannot be defined on column '"
+                                                           + columnMetadata.name + "' with type: "
+                                                           + columnMetadata.type.asCQL3Type());
+
+        for (ColumnConstraint<?> constraint : constraints)
+            constraint.validate(columnMetadata);
+    }
+
+    @Override
+    public ConstraintType getConstraintType()
+    {
+        return ConstraintType.COMPOSED;
+    }
+
+    public static class Noop extends ColumnConstraints
+    {
+        public static final Noop INSTANCE = new Noop();
+
+        public Noop()
+        {
+            super(Collections.emptyList());
+        }
+    }
+
+    public final static class Raw
+    {
+        private final List<ColumnConstraint<?>> constraints;
+
+        public Raw(List<ColumnConstraint<?>> constraints)
+        {
+            this.constraints = constraints;
+        }
+
+        public Raw()
+        {
+            this.constraints = Collections.emptyList();
+        }
+
+        public ColumnConstraints prepare()
+        {
+            return new ColumnConstraints(constraints);
+        }
+    }
+
+    public static class Serializer implements IVersionedSerializer<ColumnConstraints>
+    {
+        @Override
+        public void serialize(ColumnConstraints columnConstraint, DataOutputPlus out, int version) throws IOException
+        {
+            out.writeInt(columnConstraint.getConstraints().size());
```

Comment:
`ColumnConstraint` can have a method `getSize()` which will always return just a number which will be populated in constructor when giving constraints to that.

Noop will always return 0.